### PR TITLE
Part5

### PR DESCRIPTION
**Pull request by polypx**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -70,32 +71,33 @@ int main()
...
-    
-    std::cout << "Piano being constructed" << std::endl; //2) 
+    int counter = 0;
+    int i = 0;
+    while(i < totalKeys)
+    {
+        if(i % 12 == 1)
+        {
+        counter += 1;
+        }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -175,6 +168,18 @@ int Tree::checkSquirrelResidents()
     return numberOfSquirrels;
 }
 
+int Tree::squirrelPopulationGrowth(int initialPopulation, int numberMonths)
+{
+    numberOfSquirrels = initialPopulation;
+    for (int i = 0; i < numberMonths; i = i+3 ) 
+    {
+        numberOfSquirrels = numberOfSquirrels + numberOfSquirrels*2; // population doubles every three months
```
Comment (matkatmusic) [inline main.cpp]: This is just `numSquirrels * 3`
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -211,9 +216,12 @@ std::string City::createLaw()
...
 }
 
-int City::updatePopulation(int immigrants, int emigrants, int births, int deaths)
+int City::updatePopulation(int immigrantsYear, int emigrantsYear, int birthsYear, int deathsYear, int years)
 {
-    population = population + immigrants - emigrants + births - deaths;
+    for(int i = 0; i <= years; ++i)
+    {
+    population = population + immigrantsYear - emigrantsYear + birthsYear - deathsYear;
+    }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -261,6 +259,23 @@ int Farm::payTaxes(int totalProfit)
...
+    int weeks = 1;
+    int maximumChickens = acreage*50; // this is the maximum chickens we can fit
+    float x;
+    x = static_cast<float>(chickens);
+    while (chickens < maximumChickens)
+    { 
+    x = x + x*percentageIncreaseWeek;
+    chickens = static_cast<int>(x);    
+    ++ weeks; 
+    }
```
Comment (matkatmusic) [inline main.cpp]: Fix this indentation
Casting has not been covered yet
add space around `*` to make the code more readible
- [x] revise

File: main.cpp [hunk 5]
```diff
@@ -422,13 +453,42 @@ struct Computer
...
+int Computer::hoursTillComputerCrash(bool runningProTools)
+{
+    int hours = 0;
+    float heat = 1.1f;
+    float numberOfPlugins = 1.f;
+    float willCrash = 0.3f;
+    while(willCrash < 1.0f )
+    {
+        willCrash = willCrash * heat * numberOfPlugins;
+        numberOfPlugins = numberOfPlugins*1.2f;
```
Comment (matkatmusic) [inline main.cpp]: add space around `*` to make the code more readible
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 2 above)
Comment (polypx) [reply main.cpp]: ah ha! I simplified the population growth math

File: main.cpp (hunk 4 above)
Comment (polypx) [reply main.cpp]: removed the casting, but now we have compiler warnings (I actually only found out about casting after googling the float>int warnings  )

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**